clients/drcachesim/analyzer.cpp

```diff
@@ -471,6 +471,29 @@ analyzer_tmpl_t<RecordType, ReaderType>::process_tasks(analyzer_worker_data_t *w
 
     for (int i = 0; i < num_tools_; ++i)
         user_worker_data[i] = tools_[i]->parallel_worker_init(worker->index);
+
+    auto process_shard_exit = [=](int shard_index) {
```

Comment:
Instead of the by-value capture clause, maybe take specific data by ref? Feels safer.

Also, curious why we made this a lambda instead of a simple routine?

Reply:
Just felt like an inner function.  Moved it to a top-level helper.